crates/sui-types/src/effects/effects_v2.rs

```diff
@@ -344,11 +353,21 @@ impl TransactionEffectsV2 {
     ) -> Self {
         let unchanged_shared_objects = shared_objects
             .into_iter()
-            .filter_map(|(id, version, digest)| {
-                if changed_objects.contains_key(&id) {
-                    None
-                } else {
-                    Some((id, UnchangedSharedKind::ReadOnlyRoot((version, digest))))
+            .filter_map(|shared_input| match shared_input {
+                SharedInput::Existing((id, version, digest)) => {
+                    if changed_objects.contains_key(&id) {
+                        None
+                    } else {
+                        Some((id, UnchangedSharedKind::ReadOnlyRoot((version, digest))))
+                    }
+                }
+                SharedInput::Deleted((id, version, mutable, _)) => {
+                    debug_assert!(!changed_objects.contains_key(&id));
```

Comment:
NB: a deleted shared object should never show up in the changed objects so I added in a debug assert for it here (but please verify/correct me!).